common/utilities/tests/test_logger.py

```diff
@@ -0,0 +1,124 @@
+import io
+import logging
+from unittest import TestCase
+from unittest.mock import patch, MagicMock
+from common.utilities import integration_adapters_logger as log
+
+
+class TestLogger(TestCase):
+
+    def test_dictionary_formatting(self):
+        # Tests both removing the spaces and surrounding values with quotes if needed
+        input_dict = {
+            'Key With Space': 'value with space',  # Needs quotes
+            'EasyKey': 'EasyValue',  # Doesn't need quotes
+        }
+
+        expected_output = {
+            'Key With Space': 'KeyWithSpace="value with space"',
+            'EasyKey': 'EasyKey=EasyValue'
+        }
+        output = log._format_values_in_map(input_dict)
+        self.assertEqual(output, expected_output)
+
+    @patch('sys.stdout', new_callable=io.StringIO)
+    def test_audit(self, mock_std):
+        log.load_global_log_config()
+        log.IntegrationAdaptersLogger('TES') \
+            .info('{There Will Be No Spaces Today}', {'There Will Be No Spaces Today': 'wow qwe'}, correlation_id=2)
+
+        output = mock_std.getvalue()
+        self.assertIn('CorrelationId=2', output)
+        self.assertIn('ThereWillBeNoSpacesToday="wow qwe"', output)
+
+    @patch('sys.stdout', new_callable=io.StringIO)
+    def test_format_and_write(self, mock_std):
+        log.load_global_log_config()
+        log.IntegrationAdaptersLogger()._format_and_write(
+            message="{yes} {no} {maybe}",
+            values={'yes': 'one', 'no': 'two', 'maybe': 'three'},
+            request_id=10,
+            correlation_id=5,
+            level=logging.INFO,
+            process_key_num="100"
+        )
+
+        output = mock_std.getvalue()
+
+        # Check that each value has spaces
+        self.assertIn(' CorrelationId=5 ', output)
+        self.assertIn(' LogLevel=INFO ', output)
+        self.assertIn(' ProcessKey=SYS100', output)
+        self.assertIn(' RequestId=10 ', output)
+        self.assertIn(' maybe=three ', output)
+        self.assertIn(' yes=one ', output)
+        self.assertIn(' pid=', output)
+
+    @patch('sys.stdout', new_callable=io.StringIO)
+    def test_format_and_write_empty_vals(self, mock_std):
+        log.load_global_log_config()
+        log.IntegrationAdaptersLogger()._format_and_write(
+            message="{yes} {no} {maybe}",
+            values={'yes': 'one', 'no': 'two', 'maybe': 'three'},
+            request_id=None,
+            correlation_id=None,
+            level=logging.INFO,
+            process_key_num="100"
+        )
+
+        output = mock_std.getvalue()
+        self.assertIn(' LogLevel=INFO ', output)
+        self.assertIn(' ProcessKey=SYS100', output)
+        self.assertIn(' maybe=three ', output)
+        self.assertIn(' pid=', output)
+        self.assertIn(' yes=one ', output)
```

Comment:
Probably worth a couple of assertions to verify that the `RequestId` and `CorrelationId` keys are not present in the string.